commands/command_prune.go

```diff
@@ -0,0 +1,460 @@
+package commands
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/github/git-lfs/vendor/_nuts/github.com/rubyist/tracerx"
+
+	"github.com/github/git-lfs/git"
+	"github.com/github/git-lfs/lfs"
+	"github.com/github/git-lfs/vendor/_nuts/github.com/spf13/cobra"
+)
+
+var (
+	pruneCmd = &cobra.Command{
+		Use:   "prune",
+		Short: "Deletes old LFS files from the local store",
+		Run:   pruneCommand,
+	}
+	pruneDryRunArg      bool
+	pruneVerboseArg     bool
+	pruneVerifyArg      bool
+	pruneDoNotVerifyArg bool
+)
+
+func pruneCommand(cmd *cobra.Command, args []string) {
+
+	// Guts of this must be re-usable from fetch --prune so just parse & dispatch
+	if pruneVerifyArg && pruneDoNotVerifyArg {
+		Exit("Cannot specify both --verify-remote and --no-verify-remote")
+	}
+
+	verify := !pruneDoNotVerifyArg &&
+		(lfs.Config.FetchPruneConfig().PruneVerifyRemoteAlways || pruneVerifyArg)
+
+	prune(verify, pruneDryRunArg, pruneVerboseArg)
+
+}
+
+type PruneProgressType int
+
+const (
+	PruneProgressTypeLocal  = PruneProgressType(iota)
+	PruneProgressTypeRetain = PruneProgressType(iota)
+	PruneProgressTypeVerify = PruneProgressType(iota)
+)
+
+// Progress from a sub-task of prune
+type PruneProgress struct {
+	Type  PruneProgressType
```

Comment:
I would name this variable ProgressType.  'Type' makes me think of object types.